cnd/src/connector.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{
+    network::{DialInformation, Network},
+    seed::Seed,
+    swap_protocols::{
+        self,
+        asset::Asset,
+        metadata_store::{self, InMemoryMetadataStore, MetadataStore},
+        rfc003::{
+            self,
+            alice::{RequestError, SendRequest},
+            state_machine::SwapStates,
+            state_store::{self, InMemoryStateStore, StateStore},
+            ActorState, CreateLedgerEvents, Ledger,
+        },
+        LedgerConnectors, Metadata, SwapId,
+    },
+};
+use futures::{sync::oneshot::Sender, Future};
+use libp2p::PeerId;
+use libp2p_comit::frame::Response;
+use std::sync::Arc;
+
+/// Connector is used to connect incoming messages from the HTTP API with logic
+/// that triggers outgoing messages on the libp2p layer.
+#[derive(Debug)]
+pub struct Connector<S> {
+    pub deps: Arc<Dependencies>,
+    pub swarm: Arc<S>, // S is the libp2p Swarm within a mutex.
+}
+
+impl<S> Clone for Connector<S> {
+    fn clone(&self) -> Self {
+        Self {
+            deps: Arc::clone(&self.deps),
+            swarm: Arc::clone(&self.swarm),
+        }
+    }
+}
+
+/// Dependencies that are needed by both the libp2p network layer and the HTTP
+/// API layer.
+#[derive(Debug)]
```

Comment:
Same as above. With this you won't need to implement `Debug` on `LedgerConnectors`, which I think would also be useless.

Reply:
Nice, I did not think of this angle when writing the structs, thanks.  All removed as suggested.